### PR TITLE
Lua explorer update

### PR DESCRIPTION
**Pull request by KingoRodriguo**
- Fixed a typo in a file name : `LuaExporerTab.lua` instead of `LuaExplorerTab.lua`

- Added a new `Mods` tab in the `Lua Explorer` that lists all active mods alphabetically and gives the user the ability to reload an entire mod folder in one go.

- Base game files are omitted to prevent reloading all game files at once and breaking the game. 

- Detects all loaded mods, including those in `Zomboid/mods`, `Zomboid/Workshop`, and the Steam Workshop.

- Mod files path are stored in the `item` field of the corresponding line as `"path1 .."|".. path2 .."|".. path3"` to prevent the usage of another save file. Those paths are parsed when reloading a mod

- Change `self.fileList:setOnMouseDoubleClick()` to accomodate new feature
    - use `self.onMouseDoubleClickFile` in `FAV` and `ALL` tabs
    - use `self.onButtonMod` in `MODS` tab (Same as the buttons in the `MODS` tab)

- `self.onButtonMod` function:
    - Retrive the paths of the mod files from the `item` field of the corresponding line.
    - Parse the paths to `paths = {path1, path2, path3}`
    - Calls `reloadLuaFile` for each file in the paths table

### CODE REVIEW

File: Contents/mods/!pz_debugTools/media/lua/client/LuaExplorerPanel/LuaExplorer.lua [hunk 1]
```diff
@@ -38,6 +38,11 @@ function AUDLuaExplorer:initialise()
     AUD.LuaExplorer.favTab:initialise("FAV")
     AUD.luaExplorerWindow.nested:addView("Favorite", AUD.LuaExplorer.favTab)
 
+    -- Mods tab
```
Comment (KingoRodriguo) [inline Contents/mods/!pz_debugTools/media/lua/client/LuaExplorerPanel/LuaExplorer.lua:41]: New tab

File: Contents/mods/!pz_debugTools/media/lua/client/LuaExplorerPanel/LuaExplorerTab.lua [hunk 2]
```diff
@@ -28,7 +28,13 @@ function AUDLuaExplorerTab:initialise(category)
     self.fileList.anchorRight = true;
     self.fileList.anchorBottom = true;
     self.fileList:initialise();
-    self.fileList:setOnMouseDoubleClick(self, CustomLuaFileExplorerList.onMouseDoubleClickFile);
```
Comment (KingoRodriguo) [inline Contents/mods/!pz_debugTools/media/lua/client/LuaExplorerPanel/LuaExplorerTab.lua:31]: Move `self.fileList:setOnMouseDoubleClick()` after a check for `self.fileCategory`
- FAV and ALL Tabs reload a single file per double click
- MOD tab reload an entire mod folder

File: Contents/mods/!pz_debugTools/media/lua/client/LuaExplorerPanel/LuaExplorerTab.lua [hunk 3]
```diff
@@ -38,13 +44,15 @@ function AUDLuaExplorerTab:initialise(category)
     self.textEntry:setClearButton(true)
     self.textEntry:setText("");
 
-    if self.fileCategory == "FAV" then
```
Comment (KingoRodriguo) [inline Contents/mods/!pz_debugTools/media/lua/client/LuaExplorerPanel/LuaExplorerTab.lua:41]: Changed for accommodating the new tab

File: Contents/mods/!pz_debugTools/media/lua/client/LuaExplorerPanel/LuaExplorerTab.lua [hunk 4]
```diff
@@ -59,8 +67,10 @@ function AUDLuaExplorerTab:initialise(category)
     local button
     if self.fileCategory == "ALL" then
         button = ISButton:new(0, 0, 50, self.fileList.itemheight, "+", self, self.onButtonAdd)
-    else
```
Comment (KingoRodriguo) [inline Contents/mods/!pz_debugTools/media/lua/client/LuaExplorerPanel/LuaExplorerTab.lua:62]: Replaced the `else` statement with `elseif` to ensure that each button can only be set in its respective tab

**Review by Chuckleberry-Finn — APPROVED** [review]

**Pull request merged**